CGO45 - Popup Information Bug

DBatistaFerreira opened the pull request:
#45

* Added Department and Services to Concordia Constants
* Fixes Buildings without popup information

DBatistaFerreira committed:
* add departments and services to concordia building model

DBatistaFerreira committed:
* add departments and services to concordia building model

DBatistaFerreira committed:
* adds SGW Services and Departments in the Concordia Constants

DBatistaFerreira committed:
* adds SGW Services and Departments in the Concordia Constants

DBatistaFerreira committed:
* adds Loyola Services and Departments and in the Concordia Constants

DBatistaFerreira committed:
* adds Loyola Services and Departments and in the Concordia Constants

DBatistaFerreira committed:
Merge branch 'develop' into CGO45_Popup_Information_Bug

danielwiktorczyk committed:
Change LS learning square vertices:
- previously, was outlining a non-concordia building;
- now, outlines the new LS building
- (best estimation since google maps currently shows a parking lot)

danielwiktorczyk committed:
Change LS learning square vertices:
- more accurate depiction according to the concordia campus pdf

DBatistaFerreira committed:
* fixes whitespace inconsistencys

DBatistaFerreira committed:
* fixes whitespace inconsistencys

DBatistaFerreira committed:
Merge remote-tracking branch 'origin/CGO45_Popup_Information_Bug' into CGO45_Popup_Information_Bug

danielwiktorczyk committed:
Change K Annex coordinates:
- more accurate depiction according to the concordia campus pdf

danielwiktorczyk committed:
Change building location pins across Hall and Loyola for accuracy
- Note that some buildings require vertex changes, will be done in #45

danielwiktorczyk committed:
Change B, D, MI, MU Annex coordinates to match neighboring K building:
- more accurate depiction according to the concordia campus pdf

danielwiktorczyk reviewed (changes requested):
Missing UI

danielwiktorczyk committed:
Change FA Annex coordinates to match neighboring buildings

danielwiktorczyk committed:
Change GS Building coordinates for more accuracy

danielwiktorczyk committed:
Change LD Building coordinates, was not previously visible

DBatistaFerreira committed:
Merge pull request #60 from Pwnagebird/CGO#49

CGO#49

danielwiktorczyk committed:
Merge remote-tracking branch 'origin/CGO45_Popup_Information_Bug' into CGO45_Popup_Information_Bug

# Conflicts:
#	lib/utilities/concordia_constants.dart

danielwiktorczyk committed:
Updated building marker placement along Mackay street

danielwiktorczyk committed:
Added missing GA, Grey Nuns Annex building coordinates and information

MoreWaqar commented:
Has the requested change been handled? Not sure whether to follow through with review at current time

danielwiktorczyk commented:
@MoreWaqar we are waiting on the UI component before moving forward with this ticket.

neliop committed:
Relocate bloc logic from bloc to event classes
Add building pop up trigger when selecting from search

neliop committed:
Updated unit tests

MoreWaqar committed:
- Fix to catch a false positive JSON return from Google.

An instructionless JSON was returned passing the check, but containing no valid instruction.

MoreWaqar committed:
Merge remote-tracking branch 'origin/CGO45_Popup_Information_Bug' into CGO45_Popup_Information_Bug

neliop committed:
Make fields private in event and state classes

neliop committed:
Add departments and services to building info
Create short version and full version of building info with toggle buttons

MoreWaqar reviewed (approved):
Approved. Tested the main functionality and everything looks good. Code seems a lot cleaner too.

I think the UI change request is now met and merge can be done.

danielwiktorczyk commented on lib/utilities/application_constants.dart:
```diff
@@ -7,3 +7,6 @@ const Color concordiaRed = Color(0xff800206);
 const double listElementTextSize = 24.0;
 const double initialedIconSize = 42.0;
+
+double screenHeight;
```
Great!

danielwiktorczyk reviewed (changes requested):
Excellent UI addition! 
One small thing: bottom left "More". Can it be made into a similar style as the "Go" bottom right icon, maybe with a '?' info icon? 
![image](https://user-images.githubusercontent.com/24488852/77002335-b806f880-6931-11ea-98bc-e77935ba3965.png)

neliop committed:
Fix/restyle building info sheet buttons

danielwiktorczyk reviewed (approved):
Bottom left UI uses an icon now rather than a textual label, very good decision! 

This ticket looks ready to go to me 😎

![image](https://user-images.githubusercontent.com/24488852/77016018-db8b6c80-694c-11ea-98f3-2aede49b595f.png)
